### ISSUE TITLE
[Bug]: local variable 'videoID' referenced before assignment

### ISSUE THREAD

**Issue by scout719** [ThioJoe/YT-Spammer-Purge#851]
### Duplicate Issues

- [X] There are **no existing posts relating to my problem**
- [X] There are **existing posts relating to my problem, but the solution given, doesn't work for me.**

Found this issue: https://github.com/ThioJoe/YT-Spammer-Purge/issues/452
but this should be solved on 2.14.2 so I created a new one

### What happened?

Well, I was trying to report some comments but then this crash happened.
Running the program again with the same config file worked.

### Release version

2.16.9

### Steps to reproduce

1. Have this configuration to scan a channel's recent videos with these configs:
- (your_channel_id replaced with sample one)
[SpamPurgeConfig07.txt](https://github.com/ThioJoe/YT-Spammer-Purge/files/8840494/SpamPurgeConfig07.txt)
 

### What platform are you seeing this problem on?

Windows (.exe file)

### Relevant log output

```shell
Scanning 1/8: My twin brother has a terrible idea

Traceback (most recent call last):
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1674, in <module>
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1651, in main
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1208, in primaryInstance
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1191, in scan_video
  File "Scripts\operations.py", line 155, in get_comments
UnboundLocalError: local variable 'videoID' referenced before assignment
------------------------------------------------
Error Message: local variable 'videoID' referenced before assignment

Error - Code: X-2 occurred. This is almost definitely my fault and requires patching (big bruh moment)
Please post a bug report on the GitHub issues page, and include the above error info.
Short Link: TJoe.io/bug-report
    (In the mean time, try using a previous release of the program.)
```

### Screenshots

![image](https://user-images.githubusercontent.com/4671091/172064401-c60ed7b4-4f36-4dc7-87b5-db81340bffa4.png)

**Comment by ThioJoe** [comment]
Hm it seems to work when I do it. If you try it now, does it still do it? Also I assume the video "My twin brother has a terrible idea" is the same video as the current latest video on his channel titled "Talking to myself - Part 2"?

**Comment by scout719** [comment]
> Hm it seems to work when I do it. If you try it now, does it still do it?
Yeah, I tried again right after the crash and it worked without any issue 🤷‍♂️ I also found it very stange but decided to report anyway.

I looked at the code now, and it looks like there is an access to `videoID` on line 155:
https://github.com/ThioJoe/YT-Spammer-Purge/blob/8b93b54e786e5032412979b51247e0e70c0d86d5/Scripts/operations.py#L155

However, the variable is only assigned inside the `for item in results["items"]:`, if there are no items, the variable will never be assigned. Isn't it safer to use the input parameter `scanVideoID` instead?

> Also I assume the video "My twin brother has a terrible idea" is the same video as the current latest video on his channel titled "Talking to myself - Part 2"?

Probably, he changes the titles of his videos sometimes 😂
